src/sentry/identity/slack/provider.py

```diff
@@ -67,3 +89,48 @@ def build_identity(self, data):
             "scopes": sorted(data["scope"].split(",")),
             "data": self.get_oauth_data(data),
         }
+
+
+class SlackOAuth2LoginView(OAuth2LoginView):
+    """
+    We need to customize the OAuth2LoginView in order to support passing through
+    the `user_scope` param in the request.
+
+    The `user_scope` param would not be used when requesting identity scopes.
+    """
+
+    user_scope = ""
+
+    def __init__(
+        self, authorize_url=None, client_id=None, scope=None, user_scope=None, *args, **kwargs
+    ):
+        super(SlackOAuth2LoginView, self).__init__(
+            authorize_url=authorize_url, client_id=client_id, scope=scope, *args, **kwargs
+        )
+        if user_scope is not None:
+            self.user_scope = user_scope
+
+    def get_authorize_params(self, state, redirect_uri):
+        data = {
```

Comment:
@MeredithAnya I would use `super` here so the base value of `data` is from the default implemtation